tooling/tsc-compiler.ts

```diff
@@ -43,27 +43,41 @@ const getNameAndPos = (f: ts.FunctionLike) => {
  * This transformer will wrap content of all async functions with a try/catch that helps preserve proper async stack traces
  */
 const preserveAsyncStackTracesTransformerFactory = () => {
-  const createStackTracePreservingCatchBlockStatements = (f: ts.FunctionLike): ts.Statement[] => {
+  const createStackTracePreservingCatchBlockStatements = (f: ts.SignatureDeclaration): ts.Statement[] => {
     const statements: ts.Statement[] = [];
     const addStackLine = `\\n    at <async> ${getNameAndPos(f)}`;
     const code = `if(t instanceof Error){t.stack+="${addStackLine}";throw t}const e=new Error("Thrown["+typeof t+"]"+t);e.thrown=t;throw e`;
-    statements.push(ts.createStatement(ts.factory.createIdentifier(code)));
+    statements.push(ts.factory.createExpressionStatement(ts.factory.createIdentifier(code)));
```

Comment:
Are these changes related to updated TS version, or something else?

Reply:
Yes, these functions were marked as deprecated:

<img width="556" alt="Screenshot 2023-01-02 at 20 18 21" src="https://user-images.githubusercontent.com/1062093/210267044-c3455670-9141-45f4-aa4b-14669d2f4acb.png">

No test failed after this change, so probably it should work the same as before.